src/lib/shell/commands/Dns.cpp

```diff
@@ -83,26 +83,47 @@ class DnsShellResolverDelegate : public Dnssd::DiscoverNodeDelegate, public Addr
 
     AddressResolve::NodeLookupHandle & Handle() { return mSelfHandle; }
 
+    void LogOperationalNodeDiscovered(const Dnssd::OperationalNodeData nodeData)
+    {
+        if (!nodeData.IsValid())
+        {
+            streamer_printf(streamer_get(), "DNS browse failed - not found valid services \r\n");
+            return;
+        }
+        streamer_printf(streamer_get(), "DNS browse operational succeeded: \r\n");
+        streamer_printf(streamer_get(), "   Node ID: " ChipLogFormatX64 "-" ChipLogFormatX64 "\r\n",
+                        ChipLogValueX64(nodeData.peerId.GetCompressedFabricId()), ChipLogValueX64(nodeData.peerId.GetNodeId()));
+        streamer_printf(streamer_get(), "   hasZeroTTL: %s\r\n", nodeData.hasZeroTTL ? "true" : "false");
+    }
+
     void OnNodeDiscovered(const Dnssd::DiscoveredNodeData & discNodeData) override
     {
-        if (!discNodeData.Is<Dnssd::CommissionNodeData>())
+        if (!discNodeData.Is<Dnssd::CommissionNodeData>() && !discNodeData.Is<Dnssd::OperationalNodeData>())
+        {
+            streamer_printf(streamer_get(), "DNS browse failed - invalid node info \r\n");
+            return;
+        }
+
+        if (discNodeData.Is<Dnssd::OperationalNodeData>())
         {
-            streamer_printf(streamer_get(), "DNS browse failed - not commission type node \r\n");
+            LogOperationalNodeDiscovered(discNodeData.Get<Dnssd::OperationalNodeData>());
             return;
         }
 
-        Dnssd::CommissionNodeData nodeData = discNodeData.Get<Dnssd::CommissionNodeData>();
+        auto nodeData = discNodeData.Get<Dnssd::CommissionNodeData>();
+
         if (!nodeData.IsValid())
         {
             streamer_printf(streamer_get(), "DNS browse failed - not found valid services \r\n");
             return;
         }
 
+        streamer_printf(streamer_get(), "DNS browse succeeded: \r\n");
```

Comment:
Why was this moved?

Reply:
I moved it to ensure to have common code for both node types, which was reverted later on, missed to revert this moved.
Reverted it now.